docs/src/test.md

```diff
@@ -45,3 +45,72 @@ pytest -svv forge/test/mlir/test_ops.py::test_add
+| ------------------ | ------------------------------------------------------------------------------------------ |
+| TEST_ID            | Id of a test containing test parameters                                                    |
+
+To see examples of all available tests use `test_unique` method:
+
+```sh
+pytest -svv forge/test/operators/pytorch/test_all.py::test_unique --collect-only
+```
+
+Full list of supported query criteria
+
+| Parameter          | Description                                                                                |
+| ------------------ | ------------------------------------------------------------------------------------------ |
+| OPERATORS          | List of operators                                                                          |
+
+To query subset of test plan based on a query criteria use `test_query` method:
+
+```sh
+OPERATORS=add,div FILTERS=HAS_DATA_FORMAT,QUICK DEV_DATA_FORMATS=Float16_b,Int8 MATH_FIDELITIES=HiFi4,HiFi3 RANGE=5 pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+```sh
+OPERATORS=add,div FAILING_REASONS=DATA_MISMATCH,UNSUPPORTED_DATA_FORMAT SKIP_REASONS=FATAL_ERROR RANGE=5 pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+```sh
+FAILING_REASONS=NOT_IMPLEMENTED INPUT_SOURCES=FROM_HOST pytest -svv forge/test/operators/pytorch/test_all.py::test_query --collect-only
+```
+
+Full list of supported query criteria
+
+| Parameter             | Description                                                                                   |
+| --------------------- | --------------------------------------------------------------------------------------------- |
+| OPERATORS             | List of operators                                                                             |
+| FILTERS               | List of labmda filters. See `test/operators/pytorch/test_all.py` for all values               |
+| INPUT_SOURCES         | Supported values: FROM_ANOTHER_OP, FROM_HOST, FROM_DRAM_QUEUE, CONST_EVAL_PASS                |
+| INPUT_SHAPES          | List of input shapes to filter                                                                |
+| DEV_DATA_FORMATS      | List of dev data formats. Full list available in `test/operators/utils/test_data.py`          |
+| MATH_FIDELITIES       | List of math fidelities. Full list available in `test/operators/utils/test_data.py`           |
+| KWARGS                | List of kwargs dictionaries to filter                                                         |
+| FAILING_REASONS       | List of failing reasons. Full list available in `test/operators/utils/failing_reasons.py`     |
+| SKIP_REASONS          | List of skip reasons. Full list available in `test/operators/utils/failing_reasons.py`        |
+| RANGE                 | Limit number of results                                                                       |
```

Comment:
Let's clarify a bit. Limit number of tests we run based on defined criteria? How to we define which range to run? E.g. single example from each filter case. 

Please, correct me if I'm missing the point of this range factor :))

Reply:
RANGE is not much important and probably not needed.

Reply:
> Let's clarify a bit. Limit number of tests we run based on defined criteria? How to we define which range to run? E.g. single example from each filter case.
> 
> Please, correct me if I'm missing the point of this range factor :))

Yes, RANGE additionally limit number of tests after applying all filters. Examples from `print_query_docs`
```
export RANGE=5
export RANGE=10,20
```